pkg/console/subresource/manifestwork/oauth-client-work.go

```diff
@@ -0,0 +1,85 @@
+package manifestwork
+
+import (
+	"errors"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+	"github.com/openshift/console-operator/pkg/api"
+	oauthclientsub "github.com/openshift/console-operator/pkg/console/subresource/oauthclient"
+	"github.com/openshift/console-operator/pkg/console/subresource/util"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/klog/v2"
+	workv1 "open-cluster-management.io/api/work/v1"
+)
+
+func DefaultManagedClusterOAuthClientManifestWork(operatorConfig *operatorv1.Console, namespace string, secret string, redirects []string) *workv1.ManifestWork {
+	manifestWork := ManagedClusterOAuthClientManifestWorkStub(namespace)
+	oauthClient := oauthclientsub.DefaultManagedClusterOauthClient(secret, redirects)
+	WithManifest(manifestWork, oauthClient)
+	util.AddOwnerRef(manifestWork, util.OwnerRefFrom(operatorConfig))
+	return manifestWork
+}
+
+func ManagedClusterOAuthClientManifestWorkStub(namespace string) *workv1.ManifestWork {
+	return &workv1.ManifestWork{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      api.ManagedClusterOauthClientManifestWork,
+			Namespace: namespace,
+			Labels:    util.LabelsForManagedClusterResources(""),
+		},
+		Spec: workv1.ManifestWorkSpec{
+			Executor: &workv1.ManifestWorkExecutor{
+				Subject: workv1.ManifestWorkExecutorSubject{
+					Type: workv1.ExecutorSubjectTypeServiceAccount,
+					ServiceAccount: &workv1.ManifestWorkSubjectServiceAccount{
+						Namespace: api.OpenShiftConsoleOperatorNamespace,
+						Name:      api.OpenShiftConsoleOperatorExecutor,
+					},
+				},
+			},
+			Workload: workv1.ManifestsTemplate{
+				Manifests: []workv1.Manifest{},
+			},
+		},
+	}
+}
+
+func WithManifest(manifestWork *workv1.ManifestWork, object runtime.Object) {
+	if &manifestWork.Spec == nil {
+		manifestWork.Spec = workv1.ManifestWorkSpec{}
+	}
+
+	if &manifestWork.Spec.Workload == nil {
+		manifestWork.Spec.Workload = workv1.ManifestsTemplate{}
+	}
+
+	if &manifestWork.Spec.Workload.Manifests == nil {
+		manifestWork.Spec.Workload.Manifests = []workv1.Manifest{}
+	}
+
+	manifest := workv1.Manifest{RawExtension: runtime.RawExtension{Object: object}}
+	manifestWork.Spec.Workload.Manifests = append(manifestWork.Spec.Workload.Manifests, manifest)
+}
+
+func GetOAuthClientSecret(manifestWork *workv1.ManifestWork) (string, error) {
```

Comment:
isnt this the same secret as the [local one](https://github.com/openshift/console-operator/blob/58ea3f2d1d678adb4c0424b345b91bf3fa24b051/pkg/console/controllers/advancedclustermanagement/managedclusteroauthclients/controller.go#L115)?

Reply:
Yes, but in the future, we should generate unique secret strings for each managed cluster, and when we do, it would be nice not to have to update this logic.

Reply:
makes sense 👍 
> we should generate unique secret strings for each managed cluster

Created a story for [tracking](https://issues.redhat.com/browse/CONSOLE-3429) the work on that.